msm/src/serialization/mod.rs

```diff
@@ -102,11 +145,83 @@ mod tests {
                     constraints.push(cst.clone())
                 }
             }
+
+            for (j, lookup) in witness_env.rangecheck4_lookups.iter().enumerate() {
```

Comment:
Probably next step, but this code that collects something from witness and passes to MVLookup can be arguably moved to `witness.rs` so that `witness_env.get_inputs` would give you the inputs directly